### PR TITLE
Book component

### PR DESCRIPTION
**Pull request by EmmanuelWLX**
### Summary
- Created a book component to be rendered as a list item
- The component was tested in app/index with an element of the json file

### Screenshots
- With image
bookPhoto

- With placeholder image
bookWithoutPhoto

### Trello card
https://trello.com/c/y3ype8nF/10-n-52-componente-libro

### CODE REVIEW

File: LibraryApp/src/app/components/Book/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { View, Text, Image } from 'react-native';
+
+import styles from './styles';
+import PropTypes from 'prop-types';
+
+function Book({ title, author, photo }) {
+  return (
+    <View style={styles.container} >
+      <Image
+        source={photo && {uri: photo} || require('../../assets/book_placeholder.png')}
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/components/Book/index.js]: I would import `'../../assets/book_placeholder.png'` as a constant and then use it here.

I would use a ternary to decide which image you will show

File: LibraryApp/src/app/components/Book/index.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        style={styles.coverImage}
+      />
+      <View style={styles.rightContainer} >
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/components/Book/index.js]: Remove extra space before `>`

File: LibraryApp/src/app/components/Book/index.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+Book.propTypes = {
+  title: PropTypes.string,
+  author: PropTypes.string,
+  photo: PropTypes.string
+};
+
+Book.defaultProps = {
+  title: 'No title',
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/components/Book/index.js]: I would use something more descriptive and friendly to the user, something like `Unknown title` and `Unknown author`

File: LibraryApp/src/app/components/Book/styles.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { StyleSheet } from 'react-native';
+import { FONT_SIZE_LARGE, FONT_SIZE_MEDIUM } from '../../../constants/dimensions';
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/components/Book/styles.js]: You can add relative imports

https://github.com/tleunen/babel-plugin-module-resolver

File: LibraryApp/src/app/components/Book/styles.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import { white } from '../../../constants/colors';
+
+const IMAGE_SIZE = 80;
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/components/Book/styles.js]: I would not use constants for numbers that are used only once. However, I like having all the constants together. I will let @WFolini decide on this

File: LibraryApp/src/app/index.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import React, { Component } from 'react';
+import { View } from 'react-native';
+
+import Book from './components/Book';
+import * as books from '../books.json';
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/index.js]: As this may be temporary, add a `TODO`. Try not to import `*`

File: LibraryApp/src/app/index.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+import styles from './styles';
+
+class App extends Component {
+
+  render() {
+    return (
+      <View style={styles.container}>
+        <Book title={books[0].title} author={books[0].author} photo={books[0].image_url}/>
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/index.js]: Add a TODO here, its weird to see that `[0]`

File: LibraryApp/src/app/styles.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { StyleSheet } from 'react-native';
+
+const PADDING_TEST = 20;
+
+export default StyleSheet.create({
+  container: {
+    padding: PADDING_TEST,
+    backgroundColor: '#FFA'
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/styles.js]: Use constant color

File: LibraryApp/src/app/styles.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import { StyleSheet } from 'react-native';
+
+const PADDING_TEST = 20;
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/styles.js]: padding test?

File: LibraryApp/src/app/index.js (hunk 6 above)
Comment (EmmanuelWLX) [reply LibraryApp/src/app/index.js]: I tried not to mutate the books.json, that's why I don't do an export in that file and needed to import as *

Another way was to use require('route/book.json')... but it made the content of the file an object and not an array as I needed :/

Comment (EmmanuelWLX) [reply LibraryApp/src/app/index.js]: I correct myself... The way that it the content of the file is imported as an array is the second way, with require :sweat_smile:

File: LibraryApp/.babelrc [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+        "@components": "./src/app/components",
+        "@config": "./src/config",
+        "@constants": "./src/constants",
+        "@redux": "./src/redux",
+        "@services": "./src/services",
+        "@utils": "./src/utils"
+      }
+    }]
+  ]
+}
```
Comment (mmamaniwolox) [inline LibraryApp/.babelrc]: Try to check always for tailing line

File: LibraryApp/.eslintrc.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+  },
+  settings: {
+    "import/resolver": {
+      "babel-module": {}
+    }
+  },
+  plugins: [
+    "module-resolver"
+  ]
+};
```
Comment (mmamaniwolox) [inline LibraryApp/.eslintrc.js]: Try to check always for tailing line

File: LibraryApp/jsconfig.json [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+        "@assets": ["./src/app/assets"],
+        "@components": ["./src/app/components"],
+        "@config": ["./src/config"],
+        "@constants": ["./src/constants"],
+        "@redux": ["./src/redux"],
+        "@services": ["./src/services"],
+        "@utils": ["./src/utils"]
+    }
+  }
+}
```
Comment (mmamaniwolox) [inline LibraryApp/jsconfig.json]: what is this file?

Comment (EmmanuelWLX) [reply LibraryApp/jsconfig.json]: this is for babel-module-resolver, in an attempt to achieve autocomplete of relative imports

Comment (EmmanuelWLX) [reply LibraryApp/jsconfig.json]: however, it only completes the folders, but not the files or other folders inside :(

**Pull request merged**